pylxd/models/instance.py

```diff
@@ -291,20 +291,32 @@ def get(cls, client, name):
         return cls(client, **response.json()["metadata"])
 
     @classmethod
-    def all(cls, client):
+    def all(cls, client, **kwargs):
         """Get all instances.
 
         Instances returned from this method will only have the name
         set, as that is the only property returned from LXD. If more
-        information is needed, `Instance.sync` is the method call
-        that should be used.
+        information is needed, setting `recursion` between 1-2 will
+        obtain additional information
         """
-        response = client.api[cls._endpoint].get()
+        recursion = kwargs.get("recursion", 0)
+
+        response = client.api[cls._endpoint].get(params={'recursion': recursion})
 
         instances = []
         for url in response.json()["metadata"]:
-            name = url.split("/")[-1]
-            instances.append(cls(client, name=name))
+            if type(url) == dict:
+                # User specified recursion so returning all data for each instance at once
+                instance = cls(client, name=url['name'])
+                for key, data in url.items():
+                    try:
+                        setattr(instance, key, data)
+                    except AttributeError:
+                        pass
+                instances.append(instance)
```

Comment:
As we are now potentially fetching more than a list of URLs, I think the variable name here is a little misleading. I would consider renaming `url` to `resource`. Or perhaps renaming `instance` to `instance_class` and renaming `url` to `instance`.